avm/res/document-db/mongo-cluster/main.bicep

```diff
@@ -32,7 +32,13 @@ param diagnosticSettings diagnosticSettingFullType[]?
 param enableTelemetry bool = true
 
 @description('Optional. Whether high availability is enabled on the node group.')
-param highAvailabilityMode bool = false
+@allowed([
+  'Disabled'
+  'SameZone'
+  'ZoneRedundant'
+  'ZoneRedundantPreferred'
+])
+param highAvailabilityMode string = 'Disabled'
```

Comment:
Might be better to set this to `ZoneRedundant`.

Actually we could / should maybe use e.g., the PostgreSQL module as a reference: https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/db-for-postgre-sql/flexible-server/main.bicep

But this can also be done in a seperate PR. Just wanted to raise it.

Reply:
It's easier to do it here :smile:. But, it will require me to update the unit tests to set zone redundancy to `disabled` intentionally since it has side-effects on deployment.

Reply:
But, the WAF-compliant unit test will still have its zone-availability mode set to intentionally.

Reply:
Hey @seesharprun, please only self-resolve comments that are 'GH suggestions'. Makes a re-review easier 😉